src/main/java/org/stellar/sdk/Sep10Challenge.java

```diff
@@ -63,20 +70,21 @@ public static Transaction newChallenge(
    * It does not verify that the transaction has been signed by the client or
    * that any signatures other than the servers on the transaction are valid. Use
    * one of the following functions to completely verify the transaction:
-   * {@link Sep10Challenge#verifyChallengeTransactionSigners(String, String, Network, String, Set)} or
-   * {@link Sep10Challenge#verifyChallengeTransactionThreshold(String, String, Network, String, int, Set)} or
-   * {@link Sep10Challenge#verifyChallengeTransactionSigners(String, String, Network, String[], Set)} or
-   * {@link Sep10Challenge#verifyChallengeTransactionThreshold(String, String, Network, String[], int, Set)} or
+   * {@link Sep10Challenge#verifyChallengeTransactionSigners(String, String, Network, String, String, Set)} or
+   * {@link Sep10Challenge#verifyChallengeTransactionThreshold(String, String, Network, String, String, int, Set)} or
+   * {@link Sep10Challenge#verifyChallengeTransactionSigners(String, String, Network, String[], String, Set)} or
+   * {@link Sep10Challenge#verifyChallengeTransactionThreshold(String, String, Network, String[], String, int, Set)} or
    *
    * @param challengeXdr    SEP-0010 transaction challenge transaction in base64.
    * @param serverAccountId Account ID for server's account.
    * @param network         The network to connect to for verifying and retrieving.
    * @param domainNames     An array of home domains, one of which is expected to be included in the first Manage Data operation's string key.
+   * @param webAuthDomain   The home domain that is expected to be included as the value of the Manage Data operation with the 'web_auth_domain' key. If no such operation is included, this parameter is not used.
    * @return {@link ChallengeTransaction}, the decoded transaction envelope and client account ID contained within.
    * @throws InvalidSep10ChallengeException If the SEP-0010 validation fails, the exception will be thrown.
    * @throws IOException                    If read XDR string fails, the exception will be thrown.
    */
-  public static ChallengeTransaction readChallengeTransaction(String challengeXdr, String serverAccountId, Network network, String[] domainNames) throws InvalidSep10ChallengeException, IOException {
+  public static ChallengeTransaction readChallengeTransaction(String challengeXdr, String serverAccountId, Network network, String[] domainNames, String webAuthDomain) throws InvalidSep10ChallengeException, IOException {
```

Comment:
In the Go SDK we need to put the webauth domain before the home domains to preserve an odd but awkward case that can break applications without them realizing, so I think there would be value in having all SDKs have that field first for consistency. But it also doesn't really matter. cc @JakeUrban

Reply:
I don't think it makes sense to try to keep function signatures consistent across languages. I'm pretty sure that ship has already sailed anyway. The change made looks good to me 👍

Reply:
👍🏻